### PR TITLE
Fix line breaks added after hyphenations by blurb.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

**Comment by serhiy-storchaka** [comment]
@terryjreedy, please take a look at IDLE entries. Seems there are issues with a bullet list.

**Comment by terryjreedy** [comment]
I gather that when blurb was introduced, the existing 3.5 and 3.6 news entries were split apart into individual files and then reassembled by blurb.  In the process, it re-flowed the text, I presume with textwrap, sometimes putting newlines after hyphens. 

Why you think this is a problem?  It is standard practice in English text composition and sometimes, new hyphens are introduced between syllables.  (Textwrap is not sophisticated enough to do that.)

When Sphinx collects the x.y .rst news files into an x.y changelog, it will rewrap again, changing the locations of newlines, but again occasionally putting them after a hyphen, before justifying with whitespace.  See https://docs.python.org/3/whatsnew/changelog.html#changelog for example.

The question is whether having newlines follow - in .rst files affects the Sphinx reflow process.  Does Sphinx treat line-ending -s as optional sofe -s, eligible to be deleted along with  \n?  From my cursory reading I did not see any obvious examples of this.

In other words, I think this needed an issue and discussion as to whether to make such a change, and if so, why.

PS. The IDLE entries could stand being changed anyway, but that can be done separately and later.  An alternative to deleting the *s would be to convert them to the .rst markup.  There may be other entries that need change.

**Comment by serhiy-storchaka** [comment]
Thank you for fixing IDLE entries Terry.

Yes, the problem is that this affects the Sphinx reflow process. For example, look at the entry edited by you: I think `Sphinx- like` is incorrect writing in English, and the correct one is `Sphinx-like`. In any case inserting a space inside a long option (like in `-Werror=declaration- after-statement`) is not correct.

**Comment by terryjreedy** [comment]
'Sphinx- like' is definitely a bug.  The problem is an incompatibility between the default options of Textwrap for double reflows and a bug in its usage by blurb.

`replace_whitespace=True` replaces all white spaces, *including \n*, with ' '.
`break_on_hyphens=True` allows *insertion of \n* after '-' as an alternative to replacement of ' ' by '\n'.  So reflow 1 inserts \n, reflow 2 replaces \n with '', net result, insertion of ' ', a bug.

To avoid the problem in the future, blurb should pass `break_on_hyphens=False`.  Your patch has the effect of doing this.  If you have not manually done the backports yet, you could fix blurb and rebuild the files as an alternative.

**Comment by serhiy-storchaka** [comment]
This will not help because lines already were broken on hyphens.

**Comment by terryjreedy** [comment]
If people, including me, were breaking on hyphens when writing blurbs, they should not at Textwrap is currently written.  It seems to me that in text, replace_whitespace should replace '-\n' with '-' rather than '- '.
